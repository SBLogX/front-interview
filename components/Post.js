import React from "react";
import styled from "styled-components";
import { format, parseISO } from "date-fns";

// preferably set as env var
const POSTS_URL = "https://upply-interview.herokuapp.com";

const Root = styled.div`
  margin-top: 3rem;
`;

const Title = styled.h3`
  a {
    color: rgb(255, 94, 94);
  }
`;

const Content = styled.p`
  white-space: pre-line;
`;

const Footer = styled.div`
  margin-bottom: 1rem;
  display: flex;
  justify-content: flex-end;
  font-size: 0.8rem;
`;

const Date = styled.div``;

const Post = ({ post }) => {
  return (
    <Root>
      <img src={`${POSTS_URL}/images/${post.src}`} alt={post.title} />

      <Title>
        <a href="#">{`${post.title.substring(0, 60)}${
          post.title.length > 60 ? "..." : ""
        }`}</a>
      </Title>

      <Content>{post.text}</Content>

      <Footer>
        {post.date && (
          <Date>{format(parseISO(post.date), "dd/MM/yyyy - HH:mm")}</Date>
        )}
      </Footer>

      <hr />
    </Root>
  );
};

export { Post as default, POSTS_URL };
