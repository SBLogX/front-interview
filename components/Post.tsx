import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
    background: #fff;
    border-radius: 2px;
    display: flex;
    flex-direction: column;
    height: auto;
	margin: 1rem 0;
	padding: 1.5rem;
    position: relative;
	width: 100%;
	
	box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
	  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
	  
	img {
		object-fit: contain;
	}
`;

const Title = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-between;
	margin-bottom: 1rem;

	h3 {
		margin-bottom: 0;
		width: 500px;
		white-space: nowrap;
		overflow: hidden;
		text-overflow: ellipsis;
	}

	span {
		font-size: .7rem;
		font-weight: bold;
		min-width: 200px;
		text-align: right;
	}
`;

const Footer = ({
	title = '',
	text = '',
	date = '',
	src = '',
}) => (
    <Wrapper>
		<Title>
			<h3>{title}</h3>
			<span>{date}</span>
		</Title>
		<img
			src={`https://upply-interview.herokuapp.com/images/${src}`}
			alt={title}
		/>
		<p>{text}</p>
    </Wrapper>
);

export default Footer;
