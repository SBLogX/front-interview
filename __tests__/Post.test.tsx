import React from 'react';
import { render } from '@testing-library/react';
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import '@testing-library/jest-dom/extend-expect';

import Post from '../components/Post';

describe('Enzyme tests', () => {
    test('should render Post according to snapshot', () => {
        const wrapper = mount(<Post />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})