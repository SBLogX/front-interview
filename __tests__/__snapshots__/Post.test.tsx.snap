// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enzyme tests should render Post according to snapshot 1`] = `
<Footer>
  <Post__Wrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Post__Wrapper-sc-1djb1k0-0",
            "isStatic": true,
            "lastClassName": "dJxJxi",
            "rules": Array [
              "background:#fff;border-radius:2px;display:flex;flex-direction:column;height:auto;margin:1rem 0;padding:1.5rem;position:relative;width:100%;box-shadow:0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);transition:all 0.3s cubic-bezier(.25,.8,.25,1);img{object-fit:contain;}",
            ],
          },
          "displayName": "Post__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Post__Wrapper-sc-1djb1k0-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="Post__Wrapper-sc-1djb1k0-0 dJxJxi"
      >
        <Post__Title>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Post__Title-sc-1djb1k0-1",
                  "isStatic": true,
                  "lastClassName": "XedIj",
                  "rules": Array [
                    "display:flex;align-items:center;justify-content:space-between;margin-bottom:1rem;h3{margin-bottom:0;width:500px;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;}span{font-size:.7rem;font-weight:bold;min-width:200px;text-align:right;}",
                  ],
                },
                "displayName": "Post__Title",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Post__Title-sc-1djb1k0-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="Post__Title-sc-1djb1k0-1 XedIj"
            >
              <h3 />
              <span />
            </div>
          </StyledComponent>
        </Post__Title>
        <img
          alt=""
          src="https://upply-interview.herokuapp.com/images/"
        />
        <p />
      </div>
    </StyledComponent>
  </Post__Wrapper>
</Footer>
`;
